gunicorn/arbiter.py

```diff
@@ -473,6 +473,8 @@ def manage_workers(self):
         Maintain the number of workers by spawning or killing
         as required.
         """
+        orig_num_workers = self.num_workers
+
         if len(self.WORKERS.keys()) < self.num_workers:
```

Comment:
Thinking more on this, I think the approach you took is right, but I would not use `self.num_workers`. That is unlikely to change during this function (unless there is a concurrent signal). What is likely to change is `len(self.WORKERS)`.